### PR TITLE
[HDRP] Update decal UI

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
Fix a lot of issues (including workflow issues):
- -Y projection is now on Z to prevent workflow break with look at rotation tool
- Increased spacial position readability of the volume with handles like DensityVolume and ReflectionProbes
- Added tool edition mode to prevent handles forbidding selection in scene
- Added missing support for multi-edition (handles and materials)
- Added symmetry (alt/command) and homothety (shift) mode to the volume with handles used also in DensityVolumes and ReflectionProbes in order to maintain previous behaviour and enhance other component using these handles.
- Added scale independence to prevent metric distance to be not reliable
- Fixed an issue when entering edit mode due to API update in Unity for introduced Edit modes but also for HDProbes
- Fixed issue where migratable component version where not always serialized when part of prefab's instance
---
### Release Notes
- Added Homothety and Symetry quick edition modifier on volume used in ReflectionProbe, PlanarReflectionProbe and DensityVolume
- Added multi-edition support for DecalProjectorComponent
- Fixed "look at" edition mode of Rotation tool for DecalProjectorComponent
- Fixed issue when switching mode in ReflectionProbe and PlanarReflectionProbe
- Fixed issue where migratable component version where not always serialized when part of prefab's instance
- Updated handles for DecalProjectorComponent for enhanced spacial position readability and have edition mode for better SceneView management
- DecalProjectorComponent are now scale independent in order to have reliable metric unit (see new Size field for changing the size of the volume)

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2Fupdate-decal-ui&unity_branch=trunk&automation-tools_branch=master

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally (All graphic tests and unit tests green)
- [x] Built a player
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: 

---
### Overall Product Risks
**Technical Risk**: High

**Halo Effect**: Medium

---
### Comments to reviewers

*What is missing and cannot be done in the given amount of time:*
- An edition mode to change pivot for fine tuning rotation as in sprites. This mode should be able to manipulate pivot without affecting UV. (Seek the [TODO...] in DecalProjectorComponentEditor for adding the mode and just have to add the manipulative)
- Change the strange reset of Offset when finishing to manipulate handles. This is incompatible with keeping Offset/Pivot.
- Update API with variable visibility, variable name (m_ is for private only), in-script documentation and component name that should not contain Component for redundancy in scripts, search fields and inspector UI. 

@JordanL8 feel free to update tooltips in this PR (in DecalProjectorComponentEditor.Skin.cs)

@fredpointzero mainly added for the fix on migration in prefab instance not serialized

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorComponentEditor.Skin.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        static readonly GUIContent k_SizeContent = EditorGUIUtility.TrTextContent("Size", "Sets the size of the projector.");
+        static readonly GUIContent k_MaterialContent = EditorGUIUtility.TrTextContent("Material", "Specifies the Material this component projects as a decal.");
+        static readonly GUIContent k_DistanceContent = EditorGUIUtility.TrTextContent("Draw Distance", "Sets the distance from the Camera at which HDRP stop rendering the decal.");
+        static readonly GUIContent k_FadeScaleContent = EditorGUIUtility.TrTextContent("Start Fade", "Controls the distance from the Camera at which this component begins to fade the decal out. Expressed as a percentage of Fade Distance.");
+        static readonly GUIContent k_UVScaleContent = EditorGUIUtility.TrTextContent("Tilling", "Sets the scale for the decal Material. Scales the decal along its UV axes.");
+        static readonly GUIContent k_UVBiasContent = EditorGUIUtility.TrTextContent("Offset", "Sets the offset for the decal Material. Moves the decal along its UV axes.");
+        static readonly GUIContent k_FadeFactorContent = EditorGUIUtility.TrTextContent("Fade Factor", "In Percent");
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorComponentEditor.Skin.cs]: For the tooltip,
Controls the transparency of the decal as a percentage.

**Comment by RSlysz** [comment]
Found an issue related to manipulation in prefabs

**Comment by RSlysz** [comment]
> Can you please try making a shadergraph decal shader and see if all functionality is still supported?
I still can make a shader graph decal material and put it into a decal projector yes.

**Comment by sebastienlagarde** [comment]
Could you retarget on master ? Waiting for few test from @RemyUnity before merging.

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 2]
```diff
@@ -29,13 +31,21 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed metric distance that changed along size in DensityVolume
 - Fixed DensityVolume shape handles that have not same behaviour in advance and normal edition mode
 - Fixed normal map blending in TerrainLit by only blending the derivatives### Changed
+- Fixed multi-edition light handles and inspector shapes
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md]: clean the changelog, there is various thing that shouldn't be there

**Comment by sebastienlagarde** [comment]
Need to merge with latest master, lot of conflict here :)

**Pull request merged**